subprojects/code-quality/src/integTest/groovy/org/gradle/api/plugins/quality/codenarc/CodeNarcRelocationIntegrationTest.groovy

```diff
@@ -18,9 +18,12 @@ package org.gradle.api.plugins.quality.codenarc
 
 import org.gradle.integtests.fixtures.AbstractProjectRelocationIntegrationTest
 import org.gradle.test.fixtures.file.TestFile
+import org.gradle.util.Requires
+import org.gradle.util.TestPrecondition
 
 import static org.gradle.util.TextUtil.normaliseLineSeparators
 
+@Requires(TestPrecondition.JDK13_OR_EARLIER) // Current CodeNarc does not work with JDK14 because of the groovy version it uses
```

Comment:
If we select a different version of Groovy based on JDK, can we do away with this `@Requires`?  These "or earlier" requirements are sneaky in that they slowly remove tests as we upgrade